.tools/validate.go

```diff
@@ -0,0 +1,196 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"regexp"
+	"strings"
+)
+
+var (
+	flCommitRange = flag.String("range", "", "use this commit range instead")
+
+	validDCO = regexp.MustCompile(`^Signed-off-by: ([^<]+) <([^<>@]+@[^<>]+)>$`)
+)
+
+var DefaultRules = []ValidateRule{
+	func(c CommitEntry) error {
+		hasValid := false
+		for _, line := range strings.Split(c.Body, "\n") {
+			if validDCO.MatchString(line) {
+				hasValid = true
+			}
+		}
+		if !hasValid {
+			return fmt.Errorf("%q does not have a valid DCO", c.Commit)
+		}
+		return nil
+	},
+	// TODO add something for the cleanliness of the c.Subject
+}
+
+func main() {
+	flag.Parse()
+
+	/*
+		if len(os.Getenv("TRAVIS")) > 0 {
+			fmt.Println(os.Getenv("TRAVIS_COMMIT_RANGE"))
+			if len(os.Getenv("TRAVIS_PULL_REQUEST")) > 0 {
+				fmt.Println(os.Getenv("TRAVIS_BRANCH"))
+			}
+		}
+	*/
```

Comment:
We should probably either drop this debugging comment or uncomment it and put it behind a flag before landing this PR.